### PR TITLE
Add: Json renderer

### PR DESCRIPTION
**Pull request by JBlond**

### CODE REVIEW

**Comment by DigiLive** [comment]
I'll review this week, if still requested.

File: lib/jblond/Diff/Renderer/Text/Json.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace jblond\Diff\Renderer\Text;
+
+use jblond\Diff\Renderer\MainRendererAbstract;
+
+/**
+ * Unified diff generator for PHP DiffLib.
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/Json.php:8]: Wrong title

File: lib/jblond/Diff/Renderer/Text/Json.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ * Class Diff_Renderer_Text_Json
+ */
+class Json extends MainRendererAbstract
+{
+    /**
+     * @var array   Associative array containing the default options available for this renderer and their default
+     *              value.
+     */
+    private $subOptions = [
+        'json' => JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/Json.php:30]: 'ext-json' in composer.json is specified in 'suggest' and not in 'require' thus may be unavailable.
Perhaps keep it in 'suggest' and when using the json renderer check if the extension is loaded.

File: lib/jblond/Diff/Renderer/Text/Json.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    ];
+
+    /**
+     * InlineCli constructor.
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/Json.php:34]: Wrong title. Also in the description of the parameter.

File: lib/jblond/Diff/Renderer/Text/Json.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+     *
+     * @see Json::$subOptions
+     */
+    public function __construct(array $options = [])
+    {
+        parent::__construct($this->subOptions);
+        $this->setOptions($options);
+    }
+
+    /**
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/Json.php:46]: Incomplete docblock.

File: lib/jblond/Diff/Renderer/Text/Json.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+     * @return false|string
+     */
+    public function render()
+    {
+        $return = [];
+        $opCodes = $this->diff->getGroupedOpCodes();
+
+        foreach ($opCodes as $key => $group) {
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/Json.php:54]: Unused variable $key

File: lib/jblond/Diff/Renderer/Text/Json.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            $return[] = $this->toArray($group);
+        }
+        return json_encode($return, $this->options['json']);
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/Json.php:57]: Consider the use of JSON_THROW_ON_ERROR flag

File: lib/jblond/Diff/Renderer/Text/Json.php [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    /**
```
Comment (DigiLive) [inline lib/jblond/Diff/Renderer/Text/Json.php:60]: Incomplete docblock.

File: composer.json [hunk 8]
```diff
@@ -39,6 +39,7 @@
         "digilive/git-changelog": "^2"
     },
     "suggest": {
+        "ext-json": "*",
```
Comment (DigiLive) [inline composer.json:42]: If in suggest, check in json renderer if extension is loaded.

**Review by DigiLive — CHANGES_REQUESTED** [review]
Looks like the functionality is ok.
Left some comments, but I'm assuming you're doing a cleanup anyway.

File: lib/jblond/Diff/Renderer/Text/Json.php (hunk 2 above)
Comment (JBlond) [reply lib/jblond/Diff/Renderer/Text/Json.php]: Done

File: lib/jblond/Diff/Renderer/Text/Json.php (hunk 6 above)
Comment (JBlond) [reply lib/jblond/Diff/Renderer/Text/Json.php:63]: Good idea

File: composer.json (hunk 8 above)
Comment (JBlond) [reply composer.json:42]: :) Good idea

**Review by DigiLive — APPROVED** [review]

**Pull request merged**